lib/router/router.js

```diff
@@ -19,6 +19,13 @@ function Router(_options) {
   this.delegate = options.delegate || this.delegate;
   this.triggerEvent = options.triggerEvent || this.triggerEvent;
   this.log = options.log || this.log;
+  this.dslCallBacks = []; // NOTE: set by Ember
```

Comment:
this sounds like a short-term thing, ideally ember should track this state itself in ember-router instance, a sibling to router instance.